### PR TITLE
propagate device identity using workflows

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Issue: https://tracker.mender.io/browse/MEN-3979
When propagating device identity attributes replace point-to-point communication between deviceauth and inventory with workflows.
The "status" attribute name is reserved. If there will be an attribute called "status" in the identity attributes it will be ignored by the backend.

### CODE REVIEW

**Review by alfrunes — COMMENTED** [review]
The PR seems fine, but all the deployment integration tests are suddenly failing with off-by-one errors for some reason...

**Comment by kjaskiewiczz** [comment]
@alfrunes exactly! I've asked for the review to check if someone else can find some obvious issue with my PR, because I can't see one, and debugging the tests is taking ages

**Comment by alfrunes** [comment]
@kjaskiewiczz I've done some manual testing and everything seems fine. The only thing I can think of is if there is a race condition between the update inventory workflow and starting a dynamic deployment in the tests.

**Comment by kjaskiewiczz** [comment]
@alfrunes thanks for testing this; I was also testing this change manually and everything looks fine;
inventory (identity) update workflow shouldn't affect tests IMO, since the tests are using inventory parameters (with inventory scope), not identity
anyway, thanks again for the help and I'm going back to my investigation

File: devauth/devauth.go [hunk 1]
```diff
@@ -221,7 +257,9 @@ func (d *DevAuth) getDeviceFromAuthRequest(ctx context.Context, r *model.AuthReq
 		tenantId = idData.Tenant
 	}
 	if addDeviceErr != store.ErrObjectExists {
-		d.invClient.SetDeviceIdentity(ctx, tenantId, dev.Id, dev.IdDataStruct)
+		if err := d.setDeviceIdentity(ctx, dev, tenantId); err != nil {
+			return nil, err
```
Comment (mchalski) [inline devauth/devauth.go:261]: err != nil ... return nil?

**Comment by tranchitella** [comment]
Closing this old PR, as we addressed the issue in https://github.com/mendersoftware/deviceauth/pull/417

**Pull request closed without merging**